css/css-multicol/columnfill-auto-widows-002.html

```diff
@@ -0,0 +1,105 @@
+
+  <!--
+  Date created: November 7th 2020
+  Last modified: November 8th 2020
+  -->
+
+  <!--
+
+  Bug 1562420: Implement orphans and widows for individual columns of a multicolumn element
+  https://bugzilla.mozilla.org/show_bug.cgi?id=1562420
+
+  Bug 137367: Implement orphans and widows
+  https://bugzilla.mozilla.org/show_bug.cgi?id=137367
+
+  -->
+
+  <meta name="flags" content="">
+  <meta name="assert" content="When column boxes are filled sequentially, their content should be adjusted and fragmented in accordance with the 'widows' declaration. This test checks exactly that. In the test, since the 3rd column box was going to get only 1 line box, then a class B break point is allowed and should occur before the last line box of the 2nd column box and after the next-to-last line box of the 2nd column box so that there is a minimum of 2 line boxes at the top of the 3rd column box.">
+
+  <!--
+
+    Breaking at a class B break point is allowed only if (...)
+    the number of line boxes between the break and the end
+    of the box is the value of widows or more.
+    4.4 Unforced Breaks
+    https://www.w3.org/TR/css-break-3/#unforced-breaks
+
+  -->
+
+  <style>
+  div
+    {
+      border: orange solid 4px;
+      font-size: 20px;
+      height: 104px; /* Therefore, exactly 4 line boxes */
+      line-height: 1.3; /* computes to 26px */
+      margin-bottom: 1em;
+      padding: 0.5em; /* computes to 10px */
+      width: 490px;
+
+      columns: 3 auto;
+
+      /*
+
+      (11)  if (column-width = auto) and (column-count != auto) then
+      (12)    N := column-count;
+      (13)    W := max(0, (available-width - ((N - 1) * column-gap)) / N);
+      (14)  exit;
+
+      So:
+
+      N = 3
+
+      W = max(0, (available-width - ((N - 1) * column-gap)) / N);
+      W = max(0, (490px - ((3 - 1) * 20px)) / 3);
+      W = max(0, (490px - (2 * 20px)) / 3);
+      W = max(0, (490px - 40px) / 3);
+      W = max(0, 450px / 3);
+      W = max(0, 150px);
+      W = 150px;
+
+      */
+
+      column-fill: auto;
+
+      /*
+      'column-fill: auto'
+      Fill columns sequentially.
+      */
+
+      column-gap: 1em; /* computes to 20px */
+
+      column-rule: blue solid 4px;
+
+      widows: 2; /* widows' initial value is 2 */
+
+      /*
+      "The widows property specifies the minimum number of line boxes of a
+      block container that must be left in a fragment <em>after</em> a
+      column break <strong>at the top of a column box</strong>."
+
+      coming from
+      CSS Fragmentation Module Level 3, section 3.3 Breaks Between Lines: orphans, widows
+      https://www.w3.org/TR/css-break-3/#widows-orphans
+      */
+    }
+  </style>
+
+  <p>Test passes if the digits inside both orange-bordered rectangles are <strong>distributed identically</strong>.
+
+  <div id="test">1<br>2<br>3<br>4<br>5<br>6<br>7<br>8<br>9</div>
+
+  <div id="reference">1<br>2<br>3<br>4<br>5<br>6<br>7<br><br>8<br>9</div>
```

Comment:
I think it would make sense for the reference to use widows:1 and orphans:1.

Reply:
Yes! Excellent suggestion. I will use 'widows: 1' in the reference. 
But I do not need to use 'orphans: 1' though.